arcgispro/3D Data Co-Registration.pyt

```diff
@@ -322,17 +321,48 @@ class Register_MultiType(object):
                             "Consider reprojecting input DEM"
                             f" X = {band_description.meanCellWidth}, Y = {band_description.meanCellHeight}"
                         )
+        
+        #check that correct input 3D data types are being used ("las", "laz", "bpf", "ply", "obj", "tif", "tiff")
+        acceptable_data_list = [".las", ".laz", ".bpf", ".ply", ".obj", ".tif", ".tiff"]
+        if parameters[0].value:
+            fnd_full_path = os.fsdecode(f"{self.getLayerPath(parameters[0].valueAsText)}").replace(os.sep, "/")
+            fnd_file_extension = os.path.splitext(fnd_full_path)[-1]
+            if fnd_file_extension not in acceptable_data_list:
+                parameters[0].setErrorMessage(
+                    "File not able to be coregistered."
+                    f" Acceptable file types are {acceptable_data_list}"
+                )
+        if parameters[1].value:
+            aoi_full_path = os.fsdecode(f"{self.getLayerPath(parameters[1].valueAsText)}").replace(os.sep, "/")
+            aoi_file_extension = os.path.splitext(aoi_full_path)[-1]
+            if aoi_file_extension not in acceptable_data_list:
+                parameters[1].setErrorMessage(
+                    "File not able to be coregistered."
+                    f" Acceptable file types are {acceptable_data_list}"
+                )
 
         return
 
+    def getLayerPath(self, layer):
+        if os.path.isabs(layer):
```

Comment:
probably don't want to use `isabs` here to check if the path is absolute, you probably want to see if the path exists... but since we're not doing anything w/ this variable, we can probably restructure this entire method...


```python

def getLayerPath(self, layer):
    if not os.path.exists(layer):
        # we are working with an ArcGIS scene layer and not a file:
         desc = arcpy.Describe(layer)
         layer = os.path.join(desc.path, layer)
    return layer
```

Reply:
fixed up!